src/main/java/thaumicenergistics/common/integration/ModuleNEI.java

```diff
@@ -237,6 +239,20 @@ public ArrayList<PositionedStack> positionStacks( final ArrayList<PositionedStac
 
 	}
 
+	static class NEIGuiHandler extends INEIGuiAdapter
+	{
+		@Override
+		public boolean handleDragNDrop(GuiContainer gui, int mousex, int mousey, ItemStack draggedStack, int button)
+		{
+			if (gui instanceof ThEBaseGui && draggedStack !=null && draggedStack.getItem() !=null )
+			{
+				((ThEBaseGui)gui).draggedStack = draggedStack;
```

Comment:
Set stack size to 0, so that NEI can remove dragged stack after slot is clicked (or simply assign null)